### DISCUSSION TITLE
How to validate blank fields?

### DISCUSSION THREAD

**Discussion by notramo** [martenframework/marten#176] [Q&A] [answered]
When a form field is optional, and the user do not fill it, the form still contains the field but it's an empty string. If the model has `required: false` but `blank: false`, then its rejected.
Is there some option to convert `''` to `Nil` during validation?

**Comment by ellmetha** [comment, marked as answer]
Why set `blank: false` on your model if the corresponding field in the schema is optional? Your model field should have `blank: true` set in order to make this validation work as expected since it appears to be optional.

**Reply by notramo** [reply]
As I explained in https://github.com/martenframework/marten/issues/177#issuecomment-1962747762, the documentation was confusing, I dit not know it applies to `nil` also.
